src/compiler/msvc.rs

```diff
@@ -513,7 +513,12 @@ pub fn parse_arguments(
                     Argument::Raw(ref val) => {
                         if input_arg.is_some() {
                             // Can't cache compilations with multiple inputs.
-                            cannot_cache!("multiple input files");
+                            cannot_cache!(
+                                "multiple input files",
+                                val.clone()
+                                    .into_string()
+                                    .unwrap_or("unknown arg".to_string())
```

Comment:
:+1: 
```suggestion
                                    .unwrap_or("argument with non-UTF-8 characters".to_string())
```

Reply:
I actually replaced this with what we do in `gcc.rs`. I was under a false impression (I don't know why) that these args get bubbled up to `gcc.rs` (like the clang ones). Since they don't we need to handle them here in the same way we do in `gcc.rs`. So rather than failing on the first extra input we gather them in a list and format the whole error message after argument parsing is done.